### ISSUE TITLE
Reopening UI in Carla produces xruns

### ISSUE THREAD

**Issue by dromer** [DISTRHO/Cardinal#19]
With both the LV2 and VST3 I see up to dozens of xruns when closing and re-opening the UI.

(note that I have disabled `GLX_RGBA` in pugl)

**Comment by dromer** [comment]
On 1st gen i5 laptop running debian 11 and pipewire this is not an issue at all!

Will need to check latest build on my desktop when I get back.

**Comment by falkTX** [comment]
you need to check this yet again :)

the vcv's internal "app window" is always valid now, so there is less on the engine to do when showing the cardinal UI.

**Comment by dromer** [comment]
Still occasionally see this with the LV2, btw.

**Comment by dromer** [comment]
Just to confirm that on latest build I still see this occurring.

**Comment by falkTX** [comment]
I suspect this to be caused by lock contention, where the UI momentarily grabs locks related to the engine, but without priority inversion (the pthread read/write lock has no such APIs).

Something to fix eventually, but to be honest very low priority.

**Comment by dromer** [comment]
It doesn't happen in huge bursts as before, only after several attempts there may be a single xrun now.

Indeed not huge prio (who closes/opens their plugin UI many times in a row like this? ;) ), but at least to keep track of.

**Comment by falkTX** [comment]
Possibly fixed with aa0fbf4c36c7c41ab91ad248a1f40730e8dca37a

The entire patch was being sent from DSP -> UI, or at least attempted.
Now shouldn't happen anymore.

**Comment by falkTX** [comment]
I am closing this one as wont fix.
What we can do about it is already in place. Rack has a rwlock around the engine that needs to be taken when creating modules or changing engine stuff, which also includes setting up the widgets.
We could rework the engine lock system, but that is now at the point of not really being worth it for the amount of effort required.
